yarpl/include/yarpl/Refcounted.h

```diff
@@ -273,13 +273,26 @@ Reference<T> make_ref(Args&&... args) {
   return Reference<T>(new T(std::forward<Args>(args)...));
 }
 
-template <typename T>
+template <
+    typename T,
+    typename =
+    std::enable_if_t<std::is_base_of<Refcounted, std::decay_t<T>>::value>>
 Reference<T> get_ref(T& object) {
+  assert(
```

Comment:
Technically it's glog and not gtest, so it's not as heavyweight.

I generally dislike using `assert()` as it tends to break opt builds with that have `-Wno-unused-variable` set.